tests/unit_tests/plugin_manager_test.py

```diff
@@ -0,0 +1,219 @@
+    async def my_async_function(self, message):
+        """Async function docstring."""
+        pass
+
+    @listen_to("click_command")
+    @click.command(help="Help string for the entire function.")
+    @click.option(
+        "--option", type=int, default=0, help="Help string for the optional argument."
+    )
+    def click_commmand(self, message, option):
+        """Extended docstring.
+
+        This docstring will have 'click --help' appended
+        """
+        pass
+
+    @listen_to("hi_custom", custom="Custom attribute")
+    def hi_custom(self, message):
+        """A custom function."""
+        pass
+
+    @listen_webhook("webhook_id")
+    def webhook_listener(self, event):
+        """A webhook function."""
+        pass
+
+
+def expand_func_names(f):
+    return [f] + f.siblings
+
+
+msg_listeners = {
+    re.compile("pattern"): expand_func_names(FakePlugin.my_function),
+    re.compile("direct_pattern"): expand_func_names(FakePlugin.direct_function),
+    re.compile("another_async_pattern"): expand_func_names(
+        FakePlugin.my_async_function
+    ),
+    re.compile("async_pattern"): expand_func_names(FakePlugin.my_async_function),
+    re.compile("hi_custom"): expand_func_names(FakePlugin.hi_custom),
+    # Click commands construct a regex pattern from the listen_to pattern
+    re.compile("^click_command (.*)?"): expand_func_names(FakePlugin.click_commmand),
+}
+
+hook_listeners = {
+    re.compile("webhook_id"): expand_func_names(FakePlugin.webhook_listener)
+}
+
+
+class TestPlugin:
+    def test_init_plugin(self):
+        p = FakePlugin()
+        m = PluginManager([p])
+        with mock.patch.object(p, "initialize") as mocked:
+            m.initialize_manager(Driver(), Settings())
+
+            mocked.assert_called_once()
+
+    def test_initialize(self):
+        m = PluginManager([FakePlugin()])
+        m.initialize_manager(Driver(), Settings())
+
+        # Test whether the function was registered properly
+        assert m.message_listeners[re.compile("pattern")] == [
+            FakePlugin.my_function,
+        ]
+
+        # This function should be registered twice, once for each listener
+        assert len(m.message_listeners[re.compile("async_pattern")]) == 1
+        assert (
+            m.message_listeners[re.compile("async_pattern")][0].function
+            == FakePlugin.my_async_function.function
+        )
+
+        assert len(m.message_listeners[re.compile("another_async_pattern")]) == 1
+        assert (
+            m.message_listeners[re.compile("another_async_pattern")][0].function
+            == FakePlugin.my_async_function.function
+        )
+
+        assert len(m.webhook_listeners) == 1
+        assert m.webhook_listeners[re.compile("webhook_id")] == [
+            FakePlugin.webhook_listener
+        ]
+
+
+def create_message(
+    text="hello",
+    mentions=["qmw86q7qsjriura9jos75i4why"],
+    channel_type="O",
+    sender_name="betty",
+):
+    return Message(
```

Comment:
Why can't this be imported from the event handler test? If you prefer, we could move this to some `utils` file as well.

Reply:
Accidental leftover it's not actually used. Removed.